eng/pipelines/templates/steps/pr-matrix-presteps.yml

```diff
@@ -38,6 +38,18 @@ steps:
       ArtifactPath: '$(Build.ArtifactStagingDirectory)/PackageInfoPublishing'
 
   - pwsh: |
+      # if Azure.Core is in the list of packages we want to always keep it around for validation regardless of whether or not it is a direct package
+      Get-ChildItem -Path "$(Build.ArtifactStagingDirectory)/PackageInfo" -Filter "Azure.Core.json" -Recurse |
+        ForEach-Object {
+          $json = Get-Content $_.FullName | ConvertFrom-Json
```

Comment:
Are we at risk of long processing times when the changelist is very large, or was that more a symptom of when we were enumerating every file, and a scan/load just in directories will be quick?